bundles/org.openhab.binding.valloxmv/src/main/java/org/openhab/binding/valloxmv/internal/ValloxMVHandler.java

```diff
@@ -75,48 +75,60 @@ public void handleCommand(ChannelUID channelUID, Command command) {
-                    valloxSendSocket.request(channelUID, "0");
-                    valloxSendSocket.request(null, null);
-                } else if (OnOffType.OFF.equals(command)) {
-                    valloxSendSocket.request(channelUID, "5");
-                    valloxSendSocket.request(null, null);
+            } else if (ValloxMVBindingConstants.WRITABLE_CHANNELS_SWITCHES.contains(channelUID.getId())) {
+                if (ValloxMVBindingConstants.CHANNEL_ONOFF.equals(channelUID.getId())) {
+                    // Vallox MV MODE: Normal mode = 0, Switch off = 5
+                    strUpdateValue = (OnOffType.ON.equals(command)) ? "0" : "5";
+                } else {
+                    // Switches with ON = 1, OFF = 0
+                    strUpdateValue = (OnOffType.ON.equals(command)) ? "1" : "0";
                 }
             } else if (ValloxMVBindingConstants.WRITABLE_CHANNELS_DIMENSIONLESS.contains(channelUID.getId())) {
                 if (command instanceof QuantityType) {
                     QuantityType<Dimensionless> quantity = (QuantityType<Dimensionless>) command;
-                    valloxSendSocket.request(channelUID, Integer.toString(quantity.intValue()));
-                    valloxSendSocket.request(null, null);
+                    strUpdateValue = Integer.toString(quantity.intValue());
                 }
             } else if (ValloxMVBindingConstants.WRITABLE_CHANNELS_TEMPERATURE.contains(channelUID.getId())) {
                 if (command instanceof QuantityType) {
-                    // Convert temperature to millidegree Kelvin
+                    // Convert temperature to centiKelvin (= (Celsius * 100) + 27315 )
                     QuantityType<Temperature> quantity = ((QuantityType<Temperature>) command)
                             .toUnit(SmartHomeUnits.KELVIN);
                     if (quantity == null) {
                         return;
                     }
-                    int milliKelvin = quantity.multiply(new BigDecimal(100)).intValue();
-                    valloxSendSocket.request(channelUID, Integer.toString(milliKelvin));
-                    valloxSendSocket.request(null, null);
+                    int centiKelvin = quantity.multiply(new BigDecimal(100)).intValue();
+                    strUpdateValue = Integer.toString(centiKelvin);
                 }
+            } else {
+                // Not writable channel
+                return;
+            }
+            if (strUpdateValue != "") {
+                // Open WebSocket
+                valloxSendSocket.request(channelUID, strUpdateValue);
+                // We should wait some time to let device process the data
+                try {
+                    Thread.sleep(500);
+                } catch (InterruptedException e) {
+                    // sleep interrupted, we will get updated values during next data update cycle
+                    return;
+                }
+                valloxSendSocket.request(null, null);
```

Comment:
Any reason this couldn't be done asynchronously with a scheduler?

Reply:
To be honest with you, I'm not sure. @bjoernbrings maybe you can give some light here?  What I did was that I added 500ms sleep in between.

Reply:
Not sure why the sleep is required. Idea is to update a state and to make afterwards sure that all channels get some refresh (as a state change in one channel could affect others)

Reply:
I added sleep because in the Vallox Web GUI code (JavaScript client side code; bundle.js) there are delays in between communicating with device. However, specifically when changing one device value, Web GUI does not request full tables after changing one value - on the other hand Web GUI read tables interval is 20 seconds compared to this binding default 60 seconds. So.. I was first wondering why we need to read tables at all at this point, but on the other hand since in the original code it is made sure that there is some time in between two commands, I thought that at this stage it's better to add delay than removing read tables command. I do agree with Björn that especially if update interval is set to relatively long time, we should refresh data after modifying any value. In my view refresh should not happen immediately but like 5-10 seconds later (or faster depending what we are changing), reason being that e.g. if changing profile from Away to Boost, it takes some time for fans to adjust. How to do this in practice, I'm not sure yet. If it is possible to either trigger scheduler or modify scheduler countdown timer, that could be possible way to do this.

Reply:
You could top the refresh job and schedule a new one with an initial delay of 5 seconds.

Reply:
Thank you for the hint. If that is something that can be considered as normal way to do it I can definitely look at that. In high level looks relatively easy to do. But while I was thinking that I was wondering do we actually have bug in ValloxMVHandler.scheduleUpdates() method: is it ok to declare valloxSocket inside the method, and not at instance level?  Currently this anyway works even like this. (This may be stupid question but note that I have never done anything with Java before :) )

Reply:
You can define valloxSocket inside of a method, just make sure to do it within a try-with-resources statement so that the socket is properly closed when you are done with it.
Another consideration is that you don't want multiple sockets causing collisions on either the host device. 

But speaking of bugs I did notice in ValloxMVWebSocketListener that you aren't using your `closeLatch` properly; it never counts down so awaitClose hangs indefinitely.

Reply:
This should be now fixed. I have been running the fixed version for few days now without issues. Today I cleaned commented old code away and did some further cleanup. Have a check how does it look to you now.